src/PostgREST/App.hs

```diff
@@ -207,9 +174,11 @@ app dbstructure conf authenticator reqBody dbrole req =
         then do
           let call = B.Stmt "select " V.empty True <>
                 asJson (callProc qi $ fromMaybe M.empty (decode reqBody))
-          body :: Maybe (Identity Text) <- H.maybeEx call
+          bodyJson :: Maybe (Identity Value) <- H.maybeEx call
           return $ responseLBS status200 [jsonH]
-            (cs $ fromMaybe "[]" $ runIdentity <$> body)
+                (if hasPrefer "return=jwt"
```

Comment:
Using a return type seems interesting. Because not only it limits the functions that will return JWT, but it also documents how claims should look like.

Besides, we can provide a type in our "Sample auth system SQL script" but the user is free to tinker with different types as long as they have the same name.